rslearn/dataset/storage/file.py

```diff
@@ -0,0 +1,248 @@
+"""The default file-based dataset storage backend."""
+
+import json
+import multiprocessing
+
+import tqdm
+from upath import UPath
+
+from rslearn.dataset.window import (
+    LAYERS_DIRECTORY_NAME,
+    Window,
+    WindowLayerData,
+    get_layer_and_group_from_dir_name,
+    get_window_layer_dir,
+)
+from rslearn.log_utils import get_logger
+from rslearn.utils.fsspec import open_atomic
+from rslearn.utils.mp import star_imap_unordered
+
+from .storage import DatasetStorage, DatasetStorageFactory
+
+logger = get_logger(__name__)
+
+
+def load_window(storage: "FileDatasetStorage", window_dir: UPath) -> Window:
+    """Load the window from its directory by reading metadata.json.
+
+    Args:
+        storage: the underlying FileDatasetStorage.
+        window_dir: the path where the window is stored.
+
+    Returns:
+        the window object.
+    """
+    metadata_fname = window_dir / "metadata.json"
+    with metadata_fname.open() as f:
+        metadata = json.load(f)
+    return Window.from_metadata(storage, metadata)
+
+
+class FileDatasetStorage(DatasetStorage):
```

Comment:
Do we also need this to understand `bands.json`?

https://github.com/allenai/rslearn/blob/2cbb8665169e12e0b9c0e59d705f94b3a1862bfb/rslearn/tile_stores/default.py#L39-L40